pkg/cloudprovider/aws/suite_test.go

```diff
@@ -458,18 +497,29 @@ var _ = Describe("Allocation", func() {
 				Expect(*createFleetInput.TagSpecifications[1].ResourceType).To(Equal(ec2.ResourceTypeVolume))
 				ExpectTags(createFleetInput.TagSpecifications[1].Tags, provider.Tags)
 			})
-
 			It("should default to a generated launch template", func() {
 				pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod())[0]
 				ExpectScheduled(ctx, env.Client, pod)
-				Expect(fakeEC2API.CalledWithCreateLaunchTemplateInput.Cardinality()).To(Equal(1))
-				createLaunchTemplateInput := fakeEC2API.CalledWithCreateLaunchTemplateInput.Pop().(*ec2.CreateLaunchTemplateInput)
+
+				Expect(fakeEC2API.CalledWithCreateLaunchTemplateInput.Cardinality()).To(Equal(2))
+
+				ltNames := sets.NewString()
+				firstLt := fakeEC2API.CalledWithCreateLaunchTemplateInput.Pop().(*ec2.CreateLaunchTemplateInput)
+				ltNames.Insert(*firstLt.LaunchTemplateName)
+				ltNames.Insert(*fakeEC2API.CalledWithCreateLaunchTemplateInput.Pop().(*ec2.CreateLaunchTemplateInput).LaunchTemplateName)
+
 				Expect(fakeEC2API.CalledWithCreateFleetInput.Cardinality()).To(Equal(1))
+
 				createFleetInput := fakeEC2API.CalledWithCreateFleetInput.Pop().(*ec2.CreateFleetInput)
-				Expect(createFleetInput.LaunchTemplateConfigs).To(HaveLen(1))
 				launchTemplate := createFleetInput.LaunchTemplateConfigs[0].LaunchTemplateSpecification
-				Expect(*launchTemplate.LaunchTemplateName).To(Equal(*createLaunchTemplateInput.LaunchTemplateName))
-				Expect(createLaunchTemplateInput.LaunchTemplateData.BlockDeviceMappings[0].Ebs.Encrypted).To(Equal(aws.Bool(true)))
+				Expect(createFleetInput.LaunchTemplateConfigs).To(HaveLen(2))
```

Comment:
The test changes come since it creates two launch templates, one for non-GPU and one for GPU instances when the workload doesn't require a GPU.  I *think* this works with CreateFleet since we tell it to only launch one instance, but I'm very interested in feedback on this behavior.

Reply:
the LTs are merged and the allocation strategy is used. This will probably work 99% of the time. There could be cases for spot where a GPU instance could be provisioned since we are using the capacity-optimized allocation strategy. It would be extremely rare though.